src/Coverage.TestLogger/CodeCoverageUtility.cs

```diff
@@ -0,0 +1,151 @@
+                    stringBuilder.AppendLine("----------------");
+
+                    var header = string.Concat(
+                        string.Format(CultureInfo.CurrentCulture, CoverageResource.ModuleName),
+                        "\t\t\t",
+                        string.Format(CultureInfo.CurrentCulture, CoverageResource.NotBlocksCovered),
+                        "\t\t",
+                        string.Format(CultureInfo.CurrentCulture, CoverageResource.PercNotBlocksCovered),
+                        "\t\t",
+                        string.Format(CultureInfo.CurrentCulture, CoverageResource.BlocksCovered),
+                        "\t\t",
+                        string.Format(CultureInfo.CurrentCulture, CoverageResource.PercBlocksCovered));
+
+                    stringBuilder.AppendLine(header);
+                }
+
+                foreach (XmlNode moudle in modules)
+                {
+                    stringBuilder.AppendLine(moudle.Attributes["name"].Value + "\t\t" +
+                                                       moudle.Attributes["blocks_not_covered"].Value + "\t\t\t\t\t" +
+                                                       (100 - float.Parse(moudle.Attributes["block_coverage"].Value)) + "\t\t\t\t" +
+                                                       moudle.Attributes["blocks_covered"].Value + "\t\t\t\t" +
+                                                       moudle.Attributes["block_coverage"].Value);
+                }
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        public void AnalyzeCoverageFile(string codeCoverageFilePath, string codeCoverageRootPath)
+        {
+            try
+            {
+                var resultFile = Path.Combine(Path.GetDirectoryName(codeCoverageFilePath), Path.GetFileNameWithoutExtension(codeCoverageFilePath) + ".xml");
+                var arguments = "analyze /output:" + '"' + resultFile + '"' + " " + '"' + codeCoverageFilePath + '"';
+                string codeCoverageExe = Path.Combine(codeCoverageRootPath, CodeCoverageExeRelativePath);
+
+                if (!File.Exists(codeCoverageExe))
+                {
+                    throw new FileNotFoundException(codeCoverageExe);
+                }
+
+                Process vanguardProcess = new Process
+                {
+                    StartInfo =
+                    {
+                        UseShellExecute = false,
+                        CreateNoWindow = true,
+                        FileName = codeCoverageExe,
+                        Arguments = arguments,
+                        RedirectStandardError = true
+                    },
+                    EnableRaisingEvents = true
+                };
+
+                vanguardProcess.Exited += this.CodeCoverageExited;
+                vanguardProcess.Start();
+
+                vanguardProcess.WaitForExit();
+            }
+            catch (DllNotFoundException ex)
```

Comment:
If all the `catch` blocks need to throw, is there a reason to have them? Is some logging intended?

Reply:
Removed.